### PR TITLE
Gui: Prevent UiLoader from loading 3rd-party Qt plugins.

### PR DESCRIPTION
**Pull request by Fat-Zer**
This PR workarounds problems in QUiLoader/kde and fixes https://github.com/FreeCAD/FreeCAD/issues/8708. See the bug for details.

Due to a flaw in the QUiLoader, UiLoader were loading all designer plugins it can find in QApplication::libraryPaths(). This in general a bad practice and leads to bugs due to some plugins may perform some unexpected actions upon load which may interfere with FreeCAD's functionality.

To avoid such problems reset the libraryPaths before creation of a UiLoader object and restore afterwards.

Also:
- move setLanguageChangeEnabled(true) into constructor due to it's called every time it's being instanced anyway.
- Add a small tooltip on how to modify a shortcut.

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8810">pipeline status</a> for feature branch [PR_8810](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8810). Pipeline [801995728](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/801995728) was triggered at [773c51e](https://github.com/FreeCAD/FreeCAD/pull/8810/commits/773c51e8f2fc21dd0eb3edc57baa01cb6ef95af8). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Gui/DlgKeyboard.ui [hunk 1]
```diff
@@ -103,6 +103,9 @@
          </item>
          <item row="1" column="2" colspan="2">
           <widget class="Gui::AccelLineEdit" name="accelLineEditShortcut">
+           <property name="toolTip">
+            <string>To change a current shortcut enter the new shortcut in the field bellow and press 'Assign'.</string>
```
Comment (luzpaz) [inline src/Gui/DlgKeyboard.ui:107]: s/bellow/below/

File: src/Gui/UiLoader.cpp [hunk 2]
```diff
@@ -488,10 +489,20 @@ QString QUiLoader::errorString() const
...
     this->cw = availableWidgets();
+    setLanguageChangeEnabled(true);
+}
+
+std::unique_ptr<UiLoader> UiLoader::newInstance(QObject *parent)
+{
+    QCoreApplication *app=QCoreApplication::instance();
+    QStringList libPaths= app->libraryPaths();
+
+    app->setLibraryPaths(QStringList{}); //< backup library paths, so QUiLoader won't plugins by default
```
Comment (luzpaz) [inline src/Gui/UiLoader.cpp:501]: won't **load** plugins by default

**Comment by adrianinsaval** [comment]
doesn't this mean that shortcuts defined in kde won't work on freecad? I'm not sure if this is what we want

File: src/Gui/DlgKeyboard.ui (hunk 1 above)
Comment (Fat-Zer) [reply src/Gui/DlgKeyboard.ui]: fixed, thanks

File: src/Gui/UiLoader.cpp (hunk 2 above)
Comment (Fat-Zer) [reply src/Gui/UiLoader.cpp]: fixed, thanks

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8810">pipeline status</a> for feature branch [PR_8810](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8810). Pipeline [803062064](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/803062064) was triggered at [4cf9fd1](https://github.com/FreeCAD/FreeCAD/pull/8810/commits/4cf9fd1bd6264b826848a21d364a6b3cea22d563). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by Fat-Zer** [comment]
> doesn't this mean that shortcuts defined in kde won't work on freecad? I'm not sure if this is what we want

Nope, it doesn't... Assuming you are talking about common action shortcuts like "Open file", they are handled by a different plugin type (`platformthemes` AFAIR) this PR prevents `designer` plugins  being loaded (which provides 3rd-party widgets for dynamic ui-forms). Those plugins were disabled anyway due to [`clearPluginPaths()`][1] call, but the shared libraries were loaded into the process.

PS: The bug is a bit more specific when I initially assumed, so I've added some steps on how to reproduce to the initial report... not so much to convince to try it, but to show off that it requires pretty specific circumstances to be triggered... i.e. kde dev-libs installed on a non-kde system, switch to a python-driven workbench, concurrent shortcuts combinations (provided out of the box on the (external) BIM workbench)
PPS: sorry for a bit late answer, I thought I've posted this comment yesterday.


[1]: https://github.com/FreeCAD/FreeCAD/blob/master/src/Gui/UiLoader.cpp#L493

**Pull request merged**